### PR TITLE
[FIX] website_form: cancel records creation if error

### PR DESCRIPTION
**Pull request by adwid**
When running an automated action on record creation, if the action
raises a UserError/ValidationError, the record will still be created.

To reproduce the error:
(Need website_crm,base_automation. Enable debug mode)
1. Create an Automated Action
    - Model: Lead/Opportunity
    - Trigger: On Creation & Update
    - Trigger Fields: Phone (crm.lead)
    - Action To Do: Execute Python Code
    - Python Code:
    ```python
    if record.phone:
        raise UserError("Lorem ipsum")
    ```
2. On website, add a website form WF
    - Action: "Create an Opportunity"
3. Complete and submit WF including a phone number

Error: An error message is displayed ("Lorem ispum"). However, the lead
has been created and is present in CRM Pipeline.

An on-creation automated action is executed once the record is executed,
but nothing will cancel this records creation if the action raises a
UserError/ValidationError.

OPW-2423948

### CODE REVIEW

**Review by JKE-be — CHANGES_REQUESTED** [review]
You cannot rollback global cursor.
You don't know what you are rollback at this time, maybe other request that just your lead.

**Comment by adwid** [comment]
@JKE-be I updated the code, now the rollback will only affect the form handler

**Comment by adwid** [comment]
@rco-odoo Could you check this too ?

**Review by rco-odoo — APPROVED** [review]
Looks good to me.
@JKE-be can you check again, please?

**Comment by JKE-be** [comment]
HEllo @rco-odoo 

As explain to @adwid, my doubt was on the fact that if the creation of a record, send an email with an attachment. In case of regeneration of asset (and so auto_commit) to generate pdf, it will break de savepoint...

But it is probably a rare case. But we got it here: https://github.com/odoo/odoo/pull/24741#pullrequestreview-624301475

If it is good for you, it can be merged with a comment to explain and avoid that we remove it in the next refactoring ;)

File: addons/website_form/controllers/main.py [hunk 1]
```diff
@@ -36,8 +36,9 @@ def website_form(self, model_name, **kwargs):
             raise BadRequest('Session expired (invalid CSRF token)')
 
         try:
-            if request.env['ir.http']._verify_request_recaptcha_token('website_form'):
-                return self._handle_website_form(model_name, **kwargs)
+            with request.env.cr.savepoint():
```
Comment (rco-odoo) [inline addons/website_form/controllers/main.py:39]:
```suggestion
            # The except clause below should not let what has been done inside
            # here be committed. It should not either roll back everything in
            # this controller method. Instead, we use a savepoint to roll back
            # what has been done inside the try clause.
            with request.env.cr.savepoint():
```

**Review by rco-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**